tests/unit/CompareRaftMembers/DiffStream.js

```diff
@@ -0,0 +1,556 @@
+            },
+            {
+                desc: 'with two entries for two buckets in db and bucketd',
+                dbContents: [
+                    { key: 'bucket1/key1', value: '{}' },
+                    { key: 'bucket2/key1', value: '{}' },
+                ],
+                bucketdContents: [
+                    { key: 'bucket1/key1', value: '{}' },
+                    { key: 'bucket2/key1', value: '{}' },
+                ],
+                expectedOutput: [],
+                withDigests: [
+                    {
+                        desc: 'with no digests DB',
+                        storedDigests: null,
+                        expectedBucketStreamRequests: [
+                            {
+                                bucketName: 'bucket1',
+                                marker: null,
+                                lastKey: null,
+                            },
+                            {
+                                bucketName: 'bucket2',
+                                marker: null,
+                                lastKey: null,
+                            },
+                        ],
+                    },
+                ],
+            },
+            {
+                desc: 'with two different entries for two buckets in db and bucketd',
+                dbContents: [
+                    { key: 'bucket1/key1', value: '{"foo":"bar"}' },
+                    { key: 'bucket2/key1', value: '{"foo":"bar"}' },
+                ],
+                bucketdContents: [
+                    { key: 'bucket1/key1', value: '{"foo":"qux"}' },
+                    { key: 'bucket2/key1', value: '{"foo":"qux"}' },
+                ],
+                expectedOutput: [
+                    [
+                        { key: 'bucket1/key1', value: '{"foo":"bar"}' },
+                        { key: 'bucket1/key1', value: '{"foo":"qux"}' },
+                    ],
+                    [
+                        { key: 'bucket2/key1', value: '{"foo":"bar"}' },
+                        { key: 'bucket2/key1', value: '{"foo":"qux"}' },
+                    ],
+                ],
+                withDigests: [
+                    {
+                        desc: 'with no digests DB',
+                        storedDigests: null,
+                        expectedBucketStreamRequests: [
+                            {
+                                bucketName: 'bucket1',
+                                marker: null,
+                                lastKey: null,
+                            },
+                            {
+                                bucketName: 'bucket2',
+                                marker: null,
+                                lastKey: null,
+                            },
+                        ],
+                    },
+                ],
+            },
+            {
+                desc: 'with 7777 identical entries in db and bucketd',
+                dbContents: () => {
```

Comment:
you can use getters https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get so you don't have to use explicit checks
`if (typeof dbContents === 'function') { .. } else { .. }`